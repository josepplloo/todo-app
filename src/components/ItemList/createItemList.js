import constants from './constants';
import ceateTodoItem from '../TodoItem/createTodoItem';

import datamanager from '../../data';

import './itemList.scss';

function createitemList(itemList = datamanager.get()) {
  const itemListElement = document.createElement('UL');
  itemListElement.setAttribute('label', constants.altText);
  itemListElement.className = 'app-item-list';
  itemList.forEach((item, index) => itemListElement.appendChild(ceateTodoItem(item, index)));
  return itemListElement;
}

export default createitemList;
