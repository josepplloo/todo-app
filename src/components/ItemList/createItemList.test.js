import createItemList from './createItemList';

it('should return an <UL /> element', () => {
  const itemListElement = createItemList();
  expect(itemListElement.tagName).toBe('UL');
});
it('should have a styles and layout like the snapshot', () => {
  const itemListElement = createItemList();
  expect(itemListElement).toMatchSnapshot();
});
