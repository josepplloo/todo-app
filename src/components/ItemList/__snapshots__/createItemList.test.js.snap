// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have a styles and layout like the snapshot 1`] = `
<ul
  class="app-item-list"
  label="todo list of items"
>
  <li
    class="todo-item"
    data-key="0"
  >
    <button
      class="toggle-button"
      type="button"
    >
      <img
        src="./assets/checkmark.svg"
      />
    </button>
    <div
      class="editable-label"
      contenteditable="true"
    >
      Clean dishes
    </div>
    <button
      class="close-button"
      type="button"
    >
      <img
        src="../assets/close.svg"
      />
    </button>
  </li>
  <li
    class="todo-item"
    data-key="12"
  >
    <button
      class="toggle-button"
      type="button"
    />
    <div
      class="editable-label"
      contenteditable="true"
    >
      garbage can
    </div>
    <button
      class="close-button"
      type="button"
    >
      <img
        src="../assets/close.svg"
      />
    </button>
  </li>
  <li
    class="todo-item"
    data-key="13"
  >
    <button
      class="toggle-button"
      type="button"
    />
    <div
      class="editable-label"
      contenteditable="true"
    >
      organizer
    </div>
    <button
      class="close-button"
      type="button"
    >
      <img
        src="../assets/close.svg"
      />
    </button>
  </li>
  <li
    class="todo-item"
    data-key="14"
  >
    <button
      class="toggle-button"
      type="button"
    >
      <img
        src="./assets/checkmark.svg"
      />
    </button>
    <div
      class="editable-label"
      contenteditable="true"
    >
      teddy beaber
    </div>
    <button
      class="close-button"
      type="button"
    >
      <img
        src="../assets/close.svg"
      />
    </button>
  </li>
  <li
    class="todo-item"
    data-key="111"
  >
    <button
      class="toggle-button"
      type="button"
    />
    <div
      class="editable-label"
      contenteditable="true"
    >
      Fruit
    </div>
    <button
      class="close-button"
      type="button"
    >
      <img
        src="../assets/close.svg"
      />
    </button>
  </li>
</ul>
`;
